### PR TITLE
Warn if class variable is accessed from top level

### PR DESCRIPTION
**Pull request by bjfish**

### CODE REVIEW

File: src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.truffleruby.language.WarnNode;
+
+public class WarnTopLevelClassVariableAccessNode extends RubyBaseNode {
+
+    @Child private WarnNode warnNode = new WarnNode();
+
+    @TruffleBoundary
+    public void warnTopLevelClassVariableAccess(SourceSection sourceSection) {
+        if (sourceSection != null) {
+            warnNode.warn(sourceSection.getSource().getName(), ":", Integer.toString(sourceSection.getStartLine()),
```
Comment (eregon) [inline src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java]: RubyLanguage.fileLine :wink:

File: src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                    ": warning: class variable access from toplevel");
+        } else {
+            warnNode.warn("warning: class variable access from toplevel");
+        }
```
Comment (eregon) [inline src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java]: This makes me think `WarnNode` should handle adding the source info itself, and require a SourceSection (or null) to be passed. Can be another PR.

Comment (eregon) [reply src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java]: The `warning: ` part can also likely be shared.

File: src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java (hunk 1 above)
Comment (bjfish) [reply src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java]: Fixed.

File: src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java (hunk 2 above)
Comment (bjfish) [reply src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java]: I've refactored, please re-review.

File: src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import org.truffleruby.language.RubyBaseNode;
+import org.truffleruby.language.WarnNode;
+
+public class WarnTopLevelClassVariableAccessNode extends RubyBaseNode {
+
+    @Child private WarnNode warnNode = new WarnNode();
+
+    @TruffleBoundary
+    public void warnTopLevelClassVariableAccess(SourceSection sourceSection) {
+        warnNode.warningMessage(sourceSection, "class variable access from toplevel");
```
Comment (eregon) [inline src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java]: Given the simplicity of this node, I think we could do without now, it would only duplicate the `"class variable access from toplevel"` which seems acceptable. What do you think?

File: src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java (hunk 2 above)
Comment (eregon) [reply src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java]: Nice :)

File: src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java (hunk 3 above)
Comment (bjfish) [reply src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java]: Sounds good, fixed.
